core/runtime/src/main/java/io/quarkus/runtime/MockedThroughWrapper.java

```diff
@@ -0,0 +1,10 @@
+package io.quarkus.runtime;
+
+/**
+ * Usually, QuarkusMock mocking replaces a concrete CDI bean with a mock.
```

Comment:
In fact, we don't replace a concrete bean instance but a "delegating instance" of a client proxy. That's why only normal scopes are supported.

Reply:
I tried to add it to the javadoc :)